docs/scaling.md

```diff
@@ -12,12 +12,37 @@ As of v3.0, the controller supports having a hot-standby for [High Availability]
 
 ## Vertically Scaling
 
-You can scale the controller vertically:
+You can scale the controller vertically in these ways:
 
-- If you have many workflows, increase `--workflow-workers` and `--workflow-ttl-workers`.
-- Increase both `--qps` and `--burst`.
+### Container Resource Requests
 
-You will need to increase the controller's memory and CPU.
+If you observe the Controller using its total request CPU or memory, you should increase those.
+
+### Adding Goroutines to Increase Concurrency
+
+If you have sufficient CPU you can take advantage of it with more goroutines:
+
+- If you have many Workflows and you notice they're not being reconciled fast enough, increase `--workflow-workers`.
+- If you're using `TTLStrategy` in your Workflows and you notice they're not being deleted fast enough, increase `--workflow-ttl-workers`.
+- If you're using `PodGC` in your Workflows and you notice the Pods aren't being deleted fast enough, increase `--pod-cleanup-workers`.
+
+>= v3.5
+
+- If you're using a lot of `CronWorkflows` and they don't seem to be firing on time, increase `--cron-workflow-workers`.
+
+### K8S API Client Side Rate Limiting
+
+The K8S client library rate limits the messages that can go out. The default values are fairly low. If you frequently see a message similar to this in the Controller log (issued by the library):
+
+`Waited for 7.090296384s due to client-side throttling, not priority and fairness, request: GET:https://10.100.0.1:443/apis/argoproj.io/v1alpha1/namespaces/argo/workflowtemplates/s2t`
+
+or for >= v3.5: a warning like this (could be any CR, not just `WorkflowTemplate`):
```

Comment:
```suggestion
Or, in >= v3.5, a message similar to this (could be any CR, not just `WorkflowTemplate`):
```

- comma splices
- use same style & wording as the above for parallelism (as in grammatical parallelism, not the CPU kind 😅)

Reply:
I'd like to keep the word "warning" in there. I think the fact that it's a message at "warning" level will indicate to people that they can filter the logs to just show warning and error, and they will see this.

Reply:
Is the pre-3.5 version also a warning? Can change them to both be "warning" then to be consistent

Reply:
No, it wasn't a warning, as it wasn't our code printing the log line - it was the library printing. The motivation for adding the warning log line was so it would be more visible.

Reply:
Ohhhh, I didn't know that was an internal print -- thought it was just due to a dep upgrade that the message was changed. Thanks for the clarification!
So does that mean that _both_ are printed in 3.5 then?

Reply:
```suggestion
Or, in >= v3.5, if you see warnings similar to this (could be any CR, not just `WorkflowTemplate`):
```

I think that would be most consistent while still having the warning piece then 🤔